OpenSim/Common/DataTable.h

```diff
@@ -948,6 +948,45 @@ class DataTable_ : public AbstractDataTable {
         appendColumnLabel(columnLabel);
     }
 
+    /** Remove column corresponding to the given column index.
+
+    \throws ColumnIndexOutOfRange If the index is out of range.                  */
+        void removeColumnAtIndex(size_t index) {
+        OPENSIM_THROW_IF(isColumnIndexOutOfRange(index),
+            ColumnIndexOutOfRange,
+            index, 0, static_cast<unsigned>(_depData.ncol() - 1));
+
+        // get copy of labels
+        auto labels = getColumnLabels();
+
+        assert(labels.size() == _depData.ncol());
+
+        // shift columns unless we're already at the last column
+        for (size_t c = index; c < getNumColumns()-1; ++c) {
+            _depData.updCol((int)c) = _depData.col((int)(c + 1));
+            labels[c] = labels[c + 1];
+        }
+
```

Comment:
In general, do users care that removing a column could changes the order of other columns? If not it would be more efficient to exchange the column/label to be removed with the last column/label rather than repeated shifting.

Reply:
We encourage users to use column names, but yes I think they would be annoyed if the order changed. I agree swapping for last column would be more efficient, but it will not appear to the user that a column is simply being removed.

Reply:
Thanks @aseth1 I would just add a note to the documentation that this operation can be expensive (potentially copying all columns of the table if first data column is removed).

Reply:
My tests show that it only takes <75ms to delete 24 columns out of 25 with 50000 rows. Not sure that warrants a note.